chall03/gbudau.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+import sys
+import curses.ascii
+
+def usage():
+    sys.exit(f"usage: {sys.argv[0]} <1-9 squared_rows...>")
+
+# Check if argument count is between 1 and 20
+def check_argc(count):
+    return count > 0 and count < 21
+
+# Check if string has the same length as count and is made of ascii digits
+def isalphanum_len(string, count):
+    length = 0
+    for letter in string:
+        if not curses.ascii.isdigit(letter) or length > count:
+            return False
+        length += 1
+    return length == count
+
+# Validate all arguments
+def check_each_arg(args, count):
+    for arg in args:
+        if not isalphanum_len(arg, count):
```

Comment:
Instead of using the function `isalphanum_len()` and moving `count` around, you can:

```
if not all (len(v) == len(args[0]) and v.isdigit() for v in args):
   return False
```

Reply:
Your solution looks really good and a lot more pythonic.

I did another funcion because i was testing all/any vs or/and and found that for some reasons its not  short-circuiting when i was using a function as condition. 
So i decided to do a function that check for ASCII and length in one iteration, was probably an overkill.